examples/nlp/language_modeling/tuning/megatron_gpt_peft_eval.py

```diff
@@ -0,0 +1,157 @@
+            scaler = GradScaler(
+                init_scale=cfg.model.get("native_amp_init_scale", 2 ** 32),
+                growth_interval=cfg.model.get("native_amp_growth_interval", 1000),
+                hysteresis=cfg.model.get("hysteresis", 2),
+                enabled=False
+                if cfg.model.pipeline_model_parallel_size > 1
+                else True,  # turn off the grad scale for pipeline parallel LM model
+            )
+        if megatron_amp_o2 and not with_distributed_adam:
+            plugins.append(MegatronHalfPrecisionPlugin(precision=cfg.trainer.precision, device="cuda", scaler=scaler))
+        else:
+            plugins.append(PipelineMixedPrecisionPlugin(precision=cfg.trainer.precision, device="cuda", scaler=scaler))
+
+    if cfg.get("cluster_type", None) == "BCP":
+        plugins.append(TorchElasticEnvironment())
+
+    trainer = Trainer(plugins=plugins, strategy=strategy, **cfg.trainer)
+    peft_model_cfg = MegatronGPTPEFTModel.restore_from(
+        restore_path=cfg.model.peft.restore_from_path, trainer=trainer, return_config=True,
+    )
+
+    # hydra interpolation does not work here as the interpolation key is lost when PTL saves hparams
+    with open_dict(peft_model_cfg):
+        # update the model config of the trained model with params we want to set at inference time.
+        peft_model_cfg.precision = cfg.trainer.precision
+        peft_model_cfg.data.test_ds = cfg.model.data.test_ds
+
+    with open_dict(cfg):
+        # update the config with the trained model config
+        # required for hydra interpolation to work inside cfg.inference
+        cfg.inference.add_BOS = peft_model_cfg.data.test_ds.add_bos
+        cfg.inference.tokens_to_generate = peft_model_cfg.data.test_ds.tokens_to_generate
+
+    save_restore_connector = PEFTSaveRestoreConnector(
+        peft_model_nemo_path=cfg.model.peft.restore_from_path, peft_model_ckpt_path=None,
+    )
+    if os.path.isdir(peft_model_cfg.restore_from_path):
+        save_restore_connector.model_extracted_dir = cfg.model.restore_from_path
+    # peft_cls = _get_peft_scheme(peft_model_cfg)
+    model = NLPModel.restore_from(
+        restore_path=cfg.model.restore_from_path,
+        trainer=trainer,
+        override_config_path=peft_model_cfg,
+        save_restore_connector=save_restore_connector,
+    )
+
+    model.freeze()
+    _test_ds = model._build_dataset(peft_model_cfg.data.test_ds, is_train=False)
+    request_dl = DataLoader(
+        dataset=_test_ds[0],
+        batch_size=peft_model_cfg.data.test_ds.global_batch_size,
+        collate_fn=_test_ds[0].collate_fn,
+    )
+    config = OmegaConf.to_container(cfg.inference, resolve=True)
+    model.set_inference_config(config)
+    response = trainer.predict(model, request_dl)
+    if model.global_rank == 0:
+        print("***************************")
+        if cfg.inference.outfile_path is not None:
+            with open(cfg.inference.outfile_path, "w", encoding="utf-8") as f:
```

Comment:
add the inference server like gpt_eval script?

Reply:
Definitely want to support this. I'll add it in a separate PR.